examples/pytorch/tgn/data_preprocess.py

```diff
@@ -0,0 +1,139 @@
+import os
+
+import pandas as pd
+import numpy as np
+
+import torch
+import dgl
+
+import ssl
+from six.moves import urllib
+
+# === Below data preprocessing code are based on
+# https://github.com/twitter-research/tgn
+
+# Preprocess the raw data split each features
+
+
+def preprocess(data_name):
+    u_list, i_list, ts_list, label_list = [], [], [], []
+    feat_l = []
+    idx_list = []
+
+    with open(data_name) as f:
+        s = next(f)
+        for idx, line in enumerate(f):
+            e = line.strip().split(',')
+            u = int(e[0])
+            i = int(e[1])
+
+            ts = float(e[2])
+            label = float(e[3])  # int(e[3])
+
+            feat = np.array([float(x) for x in e[4:]])
+
+            u_list.append(u)
+            i_list.append(i)
+            ts_list.append(ts)
+            label_list.append(label)
+            idx_list.append(idx)
+
+            feat_l.append(feat)
+    return pd.DataFrame({'u': u_list,
+                         'i': i_list,
+                         'ts': ts_list,
+                         'label': label_list,
+                         'idx': idx_list}), np.array(feat_l)
+
+
+# Re index nodes for DGL convience
+def reindex(df, bipartite=True):
```

Comment:
perhaps we need a non-bipartite reindex module, like my [APAN code ](https://github.com/WangXuhongCN/APAN/blob/0474a5a5bc2729f0550d8454be6772f1ff92e8b7/preprocess/preprocess_csv.py#L104)